From: Jeff Mahoney <jeffm@suse.com>
Subject: [PATCH] psmouse: fix section type conflict

 It seems parameter functions aren't allowed to be static, or it causes
 a section type conflict. This feature isn't used very much outside
 of the standard types, which are all global and exported.

Signed-off-by: Jeff Mahoney <jeffm@suse.com>

---

 drivers/input/mouse/psmouse-base.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/drivers/input/mouse/psmouse-base.c	2007-06-12 15:45:12.000000000 -0400
+++ b/drivers/input/mouse/psmouse-base.c	2007-06-15 14:41:06.000000000 -0400
@@ -1541,7 +1541,8 @@
 }
 
 
-static int psmouse_set_maxproto(const char *val, struct kernel_param *kp)
+/* These two should be static, but it causes a section type conflict */
+int psmouse_set_maxproto(const char *val, struct kernel_param *kp)
 {
 	const struct psmouse_protocol *proto;
 
@@ -1558,7 +1559,7 @@
 	return 0;
 }
 
-static int psmouse_get_maxproto(char *buffer, struct kernel_param *kp)
+int psmouse_get_maxproto(char *buffer, struct kernel_param *kp)
 {
 	int type = *((unsigned int *)kp->arg);
 
